snakeplane/helper_classes.py

```diff
@@ -151,6 +156,16 @@ def all_inputs_completed(self) -> bool:
                     [connection.completed for connection in connections]
                 ):
                     all_inputs_completed = False
+                    
+            if len(self._state_vars.optional_input_names) > 0:  # test if there are any optional anchors
```

Comment:
It seems like this `if` statement is unnecessary, if the `optional_input_names` is empty, then it won't iterate in the next `for` loop.